### ISSUE TITLE
 intersection of ray and guassians

### ISSUE THREAD

**Issue by takeshie** [r4dl/StopThePop#10]
Thank you for the great work! 
The paper said "it presents per-pixel sorted 3D Gaussian splatting." dose it means that we can get the intersection of ray and guassians for per pixel and render the color of this with these guassian intersections? I'm wonder to know how can we get these intersections and let Rasterizer returns them?
thanks a lot!

**Comment by steimich96** [comment]
Yes, we intersect the Gaussian with a ray and choose the intersection point to be at the Gaussian's maximum contribution along the ray (see Eq. (4) in the paper). If you want to return the intersection points of each Gaussian a ray intersects, you could pre-allocate a buffer and pass it to `sortGaussiansRayHierarchicalCUDA_forward`. The [`blend_function`](https://github.com/r4dl/StopThePop-Rasterization/blob/d8cad09919ff49b11be3d693d1e71fa792f559bb/cuda_rasterizer/stopthepop/hierarchical_render.cuh#L992)  should then already receive all the information you need: `depth` (intersection point `t_opt`), `id` (Gaussian ID), and `blend_data.contributor` (offset for this pixel/ray). This way, they are already in order of closest to the camera to furthest away (Note that since our sort is still only approximate, they can potentially be slightly out of order). You just need to calculate an offset for each pixel/ray.
